### ISSUE TITLE
Support HTTPScaledObject scoped timeout

### ISSUE THREAD

**Issue by JorTurFer** [kedacore/http-add-on#813]
### Proposal

Currently, there is a timeout configuration for setting the request timeout.
Allowing HTTPScaledObjects to override the timeout for specific route can help to users who have some slow routes but they prefer to not increase the global timeout for all the routes just due to some of them

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by oto-macenauer-absa** [comment]
Hi, is this expected in future releases? 

We're using shared k8s cluster so it would be easier to use scoped timeout than asking platform team to come with values that fit all teams sharing the cluster. Also what kinds of timeouts is this expected to support?

**Comment by JorTurFer** [comment]
> Hi, is this expected in future releases?

No, this has been closed automatically due to inactivity. Are you willing to contribute with it?

> We're using shared k8s cluster so it would be easier to use scoped timeout than asking platform team to come with values that fit all teams sharing the cluster. Also what kinds of timeouts is this expected to support?

The original proposal was to override the "cold start timeout"

**Comment by oto-macenauer-absa** [comment]
> No, this has been closed automatically due to inactivity. Are you willing to contribute with it?

Sorry, I don't have any experience with go and this doesn't seem like easy task to start. 

> The original proposal was to override the "cold start timeout"

Ok, we got API that can take long to answer so I was looking for something to override the `responseHeaderTimeout`

**Comment by JorTurFer** [comment]
I've checked and I guess that we could configure all the timeouts at same place from the HTTPScaledObject, so maybe could support all of them.
https://github.com/kedacore/http-add-on/blob/main/interceptor/proxy_handlers.go#L46-L86

@tomkerkhove ?

**Comment by tpfau** [comment]
Having something like this would be great. We were trying to deploy LLMs this way on a cluster, having at most a few requests pending before adding new resources. 
But the responses (at least non streamed ones) can take quite some time before they are ready and those are then killed by the 500ms response header timeout and updating this for individual endpoints/pods would be great.

**Comment by tpfau** [comment]
I would be willing to have a closer look at this but I have literally no idea where to start (and no experience whatsoever with coding for keda). Any suggestion or example of a variable that has similar behaviors where I could have a look to start it ?

**Comment by harsh-viradia** [comment]
I found a way to change the default timeout of keda http addon, while installing http addon via the helm we can change the default value from values.yaml file.

interceptor:
  responseHeaderTimeout: 300s
  tcpConnectTimeout: 300s
  keepAlive: 300s
  resources:
    limits:
      memory: 1Gi
    requests:
      memory: 512Mi
